### DISCUSSION TITLE
Processing system - dual stream tracking

### DISCUSSION THREAD

**Discussion by EliasVaardal** [equinor/neqsim#810] [General]
Hi,

I'm curious to know if NeqSim provides a functionality for effectively tracking individual streams while managing a dual stream input within an oil processing system.

In my current process setup, I have two input streams, and after a series of separators etc. I have gas- and oil outlets. Specifically, I'm interested in determining whether NeqSim allows me to quantify the contributions of Stream A and Stream B to the total oil and gas exports.

Any insights or information on this matter would be greatly appreciated. Thank you!

**Comment by EvenSol** [comment]
Hi.
That's an important functionality that I don't' think we have a good solution for in neqsim at the moment.
I will have a deeper look at it. We once worked on giving component alias names, like methane_well1, methane_well2.
But it is still not fully working I believe.

When we have a solution for it, we will add it to the [allocation example](https://colab.research.google.com/github/EvenSol/NeqSim-Colab/blob/master/notebooks/process/allocationoilandgas2.ipynb).

**Reply by EliasVaardal** [reply]
Sounds great 👍

**Reply by EvenSol** [reply]
A temporary solution could be to do something like:

```

//well 1 components
SystemSrkEos thermoSystem = new SystemSrkEos(298.0, 10.0);
thermoSystem.addComponent("methane", 51.0);
thermoSystem.changeComponentName("methane", "methane_well1");
thermoSystem.addComponent("ethane", 1.0);
thermoSystem.changeComponentName("ethane", "ethane_well1");
thermoSystem.addComponent("propane", 1.0);
thermoSystem.changeComponentName("propane", "propane_well1");
thermoSystem.addTBPfraction("C6", 1.0, 90.0 / 1000.0, 0.7);
thermoSystem.changeComponentName("C6_PC", "C6_well1");
thermoSystem.addTBPfraction("C7", 1.0, 110.0 / 1000.0, 0.73);
thermoSystem.changeComponentName("C7_PC", "C7_well1");
thermoSystem.addTBPfraction("C8", 1.0, 120.0 / 1000.0, 0.76);
thermoSystem.changeComponentName("C8_PC", "C8_well1");
thermoSystem.addTBPfraction("C9", 1.0, 140.0 / 1000.0, 0.79);
thermoSystem.changeComponentName("C9_PC", "C9_well1");

//well 2 components
thermoSystem.addComponent("methane", 51.0);
thermoSystem.changeComponentName("methane", "methane_well2");
thermoSystem.addComponent("ethane", 1.0);
thermoSystem.changeComponentName("ethane", "ethane_well2");
thermoSystem.addComponent("propane", 1.0);
thermoSystem.changeComponentName("propane", "propane_well2");

thermoSystem.setMixingRule("classic");

thermoSystem.setMolarComposition(new double[]{90.0, 1.0, 3.0, 1.0, 1.0, 0.1, 0.1, 0.0, 0.0, 0.0 });
```

in this case all kij will be 0 (as it looks for component names in the database in the setMixingRule method call, and set it to 0 if the components are not there.

kij can however be set manually between component i and j with:

```
((PhaseEosInterface) system.getPhase(0).getMixingRule().setBinaryInteractionParameter(i,j, value);
((PhaseEosInterface) system.getPhase(1).getMixingRule().setBinaryInteractionParameter(i,j, value);
```

**Reply by EvenSol** [reply]
I have updated the allocation example with the method suggested.
